suite-common/wallet-core/src/accounts/accountsReducer.ts

```diff
@@ -425,9 +429,9 @@ export const selectIsDeviceNotEmpty = (
 ) => {
     const deviceState = selectDeviceState(state);
     const nonEmptyAccounts = selectNonEmptyDeviceAccounts(state);
-    const isDeviceDiscoveryActive = selectIsDeviceDiscoveryActive(state);
+    const hasDiscovery = selectDeviceDiscovery(state);
 
-    if ((isDeviceDiscoveryActive || !deviceState) && A.isEmpty(nonEmptyAccounts)) {
+    if ((hasDiscovery || !deviceState) && A.isEmpty(nonEmptyAccounts)) {
```

Comment:
You can have discovery but already know that the device is not empty. The method comment also claimed that. 

It would makes sense to add some short circuit evaluation `if(isNotEmpty) { return true; }`

Reply:
cc10595a14

Reply:
I was wrong, now I see that it was already working the same, just the condition wasn't so clear. Now it's easier to read but works the same, right? 👍

Reply:
It should, but you are right, this is more readable